mirobo/airpurifier.py

```diff
@@ -0,0 +1,182 @@
+from .device import Device
+from typing import Any, Dict
+import enum
+
+
+class AirPurifier(Device):
+    """Main class representing the air purifier."""
+
+    def status(self):
+        """Retrieve properties."""
+
+        # A few more properties:
+        properties = ['power', 'aqi', 'humidity', 'temp_dec',
+                      'mode', 'led', 'led_b', 'buzzer', 'child_lock',
+                      'limit_hum', 'trans_level', 'bright',
+                      'favorite_level', 'filter1_life', 'act_det',
+                      'f1_hour_used', 'use_time', 'motor1_speed']
+
+        values = self.send(
+            "get_prop",
+            properties
+        )
+        return AirPurifierStatus(dict(zip(properties, values)))
+
+    def set_mode(self, mode: OperationMode):
```

Comment:
It's better to move the class on top of the file, another possibility would be to use a forward declaration ('OperationMode'), which does not look so nice IMO and should be avoided where not really necessary (like when using it inside its own defining class).